java/ingest/ingest-runner/src/main/java/sleeper/ingest/impl/partitionfilewriter/S3AsyncClientUploader.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2022 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package sleeper.ingest.impl.partitionfilewriter;
+
+import software.amazon.awssdk.core.async.AsyncRequestBody;
+import software.amazon.awssdk.services.s3.S3AsyncClient;
+import software.amazon.awssdk.services.s3.model.PutObjectRequest;
+import software.amazon.awssdk.services.s3.model.PutObjectResponse;
+
+import java.nio.file.Path;
+import java.util.concurrent.CompletableFuture;
+
+public class S3AsyncClientUploader implements AsyncS3Uploader {
+
```

Comment:
This class appears to only be used in one test. Are there any scenarios in which we need it? Should we delete this class if we always use TransferManager in practice?

Reply:
We could. We could also use the v2 SDK version of the transfer manager which uses S3AsyncClient to begin with.

Reply:
Yes, let's do that

Reply:
If we want to use the SDK v2 version of the transfer manager that's in preview, that uses S3AsyncClient anyway. I'll probably drop this for now and try and move to that.

Reply:
I'll probably do that in a new branch.